Lib/test/test_pathlib.py

```diff
@@ -1387,6 +1387,14 @@ def test_iterdir(self):
             expected += ['linkA', 'linkB', 'brokenLink']
         self.assertEqual(paths, { P(BASE, q) for q in expected })
 
+    def test_iterdir_recursive(self):
```

Comment:
I think whatever the decision on symlink behavior, it should be tested, so an additional test with at least a structure like this: a -> b -> a

If you are going to do cycle detection, I think you also need something like this:

a
a/b
a/b/c -> a/c
a/c/b -> a/b

Presumably if you are not emitting the directories themselves, you'll also want versions that have actual files in them.

Reply:
Implemented in fbae73a